adapter/circonus/circonus_test.go

```diff
@@ -0,0 +1,307 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package circonus
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	cgm "github.com/circonus-labs/circonus-gometrics"
+
+	"io/ioutil"
+
+	"istio.io/mixer/adapter/circonus/config"
+	"istio.io/mixer/pkg/adapter"
+	"istio.io/mixer/pkg/adapter/test"
+)
+
+var validTestGauge = int64(123)
+var validTestCounter = int64(1)
+var validTestHistogramString = "H[1.4e+08]=1"
+var validTestHistogramVal = 146 * time.Millisecond
+var validTestHistogramIntVal = 3459
+var validTestHistogramIntString = "H[3.4e+03]=1"
+
+func testServer(t *testing.T) *httptest.Server {
```

Comment:
Does the test need to be this complex?
Can you define an interface with `Timing`, `Gauge`, and `Increment` functions and just test what you send to your fake rather than adding all the http details here?